x-pack/plugins/cases/public/components/user_action_tree/helpers.tsx

```diff
@@ -157,10 +157,11 @@ export const getPushedServiceLabelTitle = (action: CaseUserActions, firstPush: b
 
 export const getPushInfo = (
   caseServices: CaseServices,
-  parsedValue: { connector_id: string; connector_name: string },
+  // a JSON parse failure will result in null for parsedValue
+  parsedValue: { connector_id: string | null; connector_name: string } | null,
   index: number
 ) =>
-  parsedValue != null
+  parsedValue != null && parsedValue.connector_id != null
```

Comment:
The `connector_id` will be `null` if the reference cannot be found within the saved object. I did that instead of making the entire `external_service` field `null` to preserve the other fields in `external_service`.